libsignal-protocol-swift/Misc/Signature.swift

```diff
@@ -0,0 +1,93 @@
+//
+//  Signature.swift
+//  libsignal-protocol-swift
+//
+//  Created by Benjamin Garrigues on 06/07/2022.
+//  Copyright © 2022 User. All rights reserved.
+//
+
+import Foundation
+import SignalModule
+
+public struct Signature {
+    public enum SignatureError: Error {
+        case signError(SignalError)
+        case verifyError(SignalError)
+        case nilSignatureBuffer
+        case keyDecodeError(SignalError)
+    }
+
+    // privatekey: Taken from KeyPair
+    public static func sign(payload: Data, withPrivateKey privateKey: Data) -> Result<Data, SignatureError> {
+
+        // Convert private key to [UInt8]
+        let privateBuffer = privateKey.signalBuffer
+        defer { signal_buffer_free(privateBuffer) }
+        let privLength = signal_buffer_len(privateBuffer)
+        let privData = signal_buffer_data(privateBuffer)!
+        var privKey: OpaquePointer? = nil
+        let keyDecodeResult = withUnsafeMutablePointer(to: &privKey) {
+            curve_decode_private_point($0, privData, privLength, Signal.context)
+        }
+        guard keyDecodeResult == 0 else {
+            return .failure(.keyDecodeError(SignalError(value: keyDecodeResult)))
+        }
+        defer { ec_private_key_destroy(privKey) }
+
+        var signatureBuf: OpaquePointer? = nil
+        let signResult = withUnsafeMutablePointer(to: &signatureBuf) { signatureBufPtr in
+            return payload.withUnsafeBytes {
```

Comment:
Isn't `withUnsafeBytes` deprecated ? You should load its inner `UnsafeRawBufferPointer` as the type you want, see https://stackoverflow.com/a/55378571/2219009

Reply:
It wasn't in swift4.2 (the actual target for the project). I switch to 5 and fixed it. it was actually simpler than in the stackoverflow. in the case of byte arrays you don't even need to load (or even rebind the memory). By default it's already in the good units and there are no memory alignment issues.